File: app/src/main/java/com/doubean/ford/ui/groupSearch/GroupSearchViewModelFactory.java
package com.doubean.ford.ui.groupSearch;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.doubean.ford.data.repository.GroupRepository;

public class GroupSearchViewModelFactory implements ViewModelProvider.Factory {
    private GroupRepository groupRepository;

    public GroupSearchViewModelFactory(GroupRepository groupRepository) {
        this.groupRepository = groupRepository;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new GroupSearchViewModel(groupRepository);
    }
}

File: app/src/main/java/com/doubean/ford/ui/groups/groupSearch/GroupSearchViewModelFactory.kt
package com.doubean.ford.ui.groups.groupSearch

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.doubean.ford.data.repository.GroupRepository

class GroupSearchViewModelFactory(private val groupRepository: GroupRepository) :
    ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return GroupSearchViewModel(groupRepository) as T
    }
}